backend/inspirehep/records/marshmallow/literature/ui.py

```diff
@@ -150,6 +155,19 @@ def get_number_of_references(self, data):
         references = data.get("references")
         return self.get_len_or_missing(references)
 
+    def get_linked_book(self, data):
+        parent = get_parent_record(data)
+        if parent and "titles" in parent and "control_number" in parent:
+            endpoint = PidStoreBase.get_endpoint_from_pid_type(
+                PidStoreBase.get_pid_type_from_schema(data["$schema"])
+            )
+            endpoint_item = "invenio_records_rest.{0}_item".format(endpoint)
+            self = url_for(
```

Comment:
but why do you actually need this? don't we have `self` directly in the record?

Reply:
We have $self only if we put it there. So if we create new record (for example in tests) and we won't put $self inside data, then $self wont' be there.